src/ray/raylet/node_manager.h

```diff
@@ -615,6 +623,23 @@ class NodeManager : public rpc::NodeManagerServiceHandler {
 
   /// Map of workers leased out to direct call clients.
   std::unordered_map<int, std::shared_ptr<Worker>> leased_workers_;
+
+  /// The new resource scheduler for direct task calls.
+  std::shared_ptr<ClusterResourceScheduler> new_resource_scheduler_;
+  /// Map of leased workers to their current resource usage.
+  // std::unordered_map<int, TaskRequest> leased_worker_resources_; XXX
+  std::unordered_map<int, std::unordered_map<std::string, double>>
+      leased_worker_resources_;
+
+  typedef std::function<void(std::shared_ptr<Worker>, ClientID spillback_to,
+                             std::string address, int port)>
+      ScheduleFn;
+
+  /// Queue of lease requests that are waiting for resources to become available.
+  /// TODO this should be a queue for each SchedulingClass
+  std::deque<std::pair<ScheduleFn, Task>> new_pending_queue_;
```

Comment:
Consider renaming to `tasks_to_schedule_`.

Reply:
done